File: go/2744.go
package main

func maximumNumberOfStringPairs(words []string) (res int) {
	m := map[int]int{}
	for _, word := range words {
		x := int(word[0])*100 + int(word[1])
		rx := int(word[1])*100 + int(word[0])
		res += m[rx]
		m[x]++
	}
	return res
}

File: python/2744.py
from typing import List
class Solution:
    def maximumNumberOfStringPairs(self, words: List[str]) -> int:
        ans = 0
        s = set()
        for word in words:
            if word[::-1] in s:
                ans += 1
            s.add(word)
        return ans
